homeassistant/components/bring/sensor.py

```diff
@@ -88,16 +92,49 @@ async def async_setup_entry(
 ) -> None:
     """Set up the sensor platform."""
     coordinator = config_entry.runtime_data
-
-    async_add_entities(
-        BringSensorEntity(
-            coordinator,
-            bring_list,
-            description,
-        )
-        for description in SENSOR_DESCRIPTIONS
-        for bring_list in coordinator.data.values()
-    )
+    lists_added: set[str] = set()
+
+    @callback
+    def add_entities() -> None:
+        """Add or remove sensor entities."""
+        nonlocal lists_added
+        entities: list[BringSensorEntity] = []
+        entity_registry = er.async_get(hass)
+
+        for bring_list in coordinator.lists:
+            if bring_list["listUuid"] not in lists_added:
+                entities.extend(
+                    BringSensorEntity(
+                        coordinator,
+                        bring_list,
+                        description,
+                    )
+                    for description in SENSOR_DESCRIPTIONS
+                )
+            lists_added.add(bring_list["listUuid"])
+
+        user = {x["listUuid"] for x in coordinator.user_settings["userlistsettings"]}
+        for list_uuid in user | lists_added:
+            if any(
+                bring_list["listUuid"] == list_uuid for bring_list in coordinator.lists
+            ):
+                continue
+
+            for description in SENSOR_DESCRIPTIONS:
+                if entity_id := entity_registry.async_get_entity_id(
+                    SENSOR_DOMAIN,
+                    DOMAIN,
+                    f"{coordinator.config_entry.unique_id}_{list_uuid}_{description.key}",
+                ):
+                    entity_registry.async_remove(entity_id)
```

Comment:
We have a device per list right? Let's just remove the whole device as HA will take care of the rest